From 52747aebc16ae524743f5b9723f7c1e9bd89b957 Mon Sep 17 00:00:00 2001
From: Ankit Agrawal <ankit.agarwal@intel.com>
Date: Thu, 18 May 2023 15:45:55 +0530
Subject: [PATCH] Fixed Build Error time header file.

Build error due to duplicate defination
of struct timespec.

Guarded with #ifndef condition.

Tracked-On: OAM-106853
Signed-off-by: Tanuj Tekriwal <tanuj.tekriwal@intel.com>
---
 src/c11/time.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/c11/time.h b/src/c11/time.h
index b9f779c83d3..15e2d2ecec1 100644
--- a/src/c11/time.h
+++ b/src/c11/time.h
@@ -28,12 +28,14 @@ extern "C" {
  * So detecting `HAVE_STRUCT_TIMESPEC` in meson script dynamically.
  */
 #ifndef HAVE_STRUCT_TIMESPEC
+#ifndef _STRUCT_TIMESPEC
 struct timespec
 {
     time_t tv_sec;  // Seconds - >= 0
     long   tv_nsec; // Nanoseconds - [0, 999999999]
 };
 #endif
+#endif
 
 /*-------------------------- functions --------------------------*/
 
-- 
2.17.1

